src/Stratis.Bitcoin/Consensus/ConsensusManagerBehavior.cs

```diff
@@ -452,11 +456,21 @@ protected async Task<ConnectNewHeadersResult> PresentHeadersLockedAsync(List<Blo
             }
             catch (ConnectHeaderException)
             {
+                // This is typically thrown when the the first header refers to a previous block hash that is not in
+                // the header tree currently. This is not regarded as bannable, as it could be a legitimate reorg.
                 this.logger.LogDebug("Unable to connect headers.");
                 this.cachedHeaders.Clear();
 
-                // Resync in case can't connect.
-                await this.ResyncAsync().ConfigureAwait(false);
+                // Resync in case we can't connect the header.
+                // We do not want to do this too frequently, as it is possible for an infinite loop to arise if a node
+                // continues to supply headers that cannot be connected. In lieu of more pervasive rate limiting this
+                // quick check should reduce the bandwidth consumption of spurious resync requests.
+                // Resyncs will eventually be triggered by timer elsewhere so a peer will not remain stuck indefinitely.
+                if ((DateTime.Now - this.lastHeaderConnectResync) > TimeSpans.Second)
```

Comment:
@zeptin I spoke to @dangershony about this one and we can even increase to a minute!

Reply:
Whoops, I didn't see your comment @fassadlr - I've updated the interval to a minute now.